helix-core/src/case_conversion.rs

```diff
@@ -2,27 +2,45 @@ use crate::Tendril;
 
 // todo: should this be grapheme aware?
 
-pub fn to_pascal_case(text: impl Iterator<Item = char>) -> Tendril {
+fn to_case<I>(text: I, to_case_with: fn(I, &mut Tendril)) -> Tendril
+where
+    I: Iterator<Item = char>,
+{
     let mut res = Tendril::new();
-    to_pascal_case_with(text, &mut res);
+    to_case_with(text, &mut res);
```

Comment:
Nit: i don't like the `_with` suffix too much, I find it doesn't convey much meaning, at least in my opinion. I would much rather have it named something like 


```suggestion
    case_conversion(text, &mut res);
```

Or

```suggestion
    convert_to_case(text, &mut res);
```

Or something like that. Then the other methods could be given the respective `_with` methods as callbacks:

```rust
to_case(text, |args...| { /* Implementation... */ })
```

Reply:
All the functions which mutate in-place I've prefixed with `into`, all the ones which don't are prefixed with `to` now. please let me know what you think now!